test/lifecycle_test.dart

```diff
@@ -346,6 +358,7 @@ void sharedLifecycleTests<T extends react.Component>({
         expect(
             component.lifecycleCalls,
             equals([
+              matchCall('getChildContext'),
```

Comment:
Is this left over from a bad merge with Keal's context PR?

Reply:
It's probably left over from when he branched off my in progress context pr...
https://github.com/cleandart/react-dart/pull/160/files#diff-c00a02aa016b25a86fb30a2093f935a1R349

Reply:
Just to recap standup for future reference, it appears they're needed. The lifecycle calls are part of the actual test results, and it makes sense (to me) for them to be there based on the tests. If anyone wants to double check that, it may be a good QA check, but `getChildContext` is unrelated to `Component2` so it seems more likely that the overlap in branching created this need as opposed to adding new lifecycles to `Component2`.